source/math/Quaternion.ooc

```diff
@@ -260,6 +262,80 @@ Quaternion: cover {
 		}
 		result
 	}
+	weightedMeanOfQuaternions: static func (quaternions: VectorList<This>, weights: Float[]) -> This {
+		// Implementation of the QUEST algorithm. For variable meanings, see original publication:
+		// M.D. Shuster and S.D. Oh, "Three-Axis Attitude Determination from Vector Observations", 1981
+		// [http://www.malcolmdshuster.com/Pub_1981a_J_TRIAD-QUEST_scan.pdf]
+		// TODO: Fix singularity problem when the angle is close to PI, using sequential rotations
+		vectorCount := 3 * quaternions count
+		referenceVectors := FloatMatrix new(vectorCount, 3)
+		observationVectors := FloatMatrix new(vectorCount, 3)
+		This _createVectorMeasurementsForQuest(quaternions, referenceVectors, observationVectors)
+		This _normalizeFloatArray(weights, quaternions count)
+
+		B := FloatMatrix new(3, 3)
+		w := FloatMatrix new(1, 3)
+		v := FloatMatrix new(1, 3)
+		for (n in 0 .. vectorCount) {
+			for (i in 0 .. referenceVectors dimensions height) {
+				v set(0, i, referenceVectors get(n, i))
+				w set(0, i, observationVectors get(n, i))
+			}
+			B += weights[n / 3] / 3.0f * w * v transpose()
+		}
+		S := B + B transpose()
```

Comment:
Don't think so because it has no special properties. The product of a matrix and its transpose does, though (it's symmetric) and is used in solve(), but I can't find a name for it either. There is more to be gained there, but as you say it's still a specialization. Depends on how much performance we want out of this.